app/controllers/VatDetailsController.scala

```diff
@@ -163,19 +165,27 @@ class VatDetailsController @Inject()(val messagesApi: MessagesApi,
     }
   }
 
-  private def retrieveIsNonMTDfB(mandationStatus: HttpGetResult[MandationStatus]): Option[Boolean] = {
+  private[controllers] def retrieveIsOfStatus(mandationStatus: HttpGetResult[MandationStatus], expectedType: String*): Option[Boolean] = {
     mandationStatus.fold(
       _ => None,
-      result => Some(result.mandationStatus == nonMTDfB)
+      result => Some(expectedType match {
+        case statusSeq => statusSeq.foldRight(false) { (expectedMandation, currentStatus) =>
+          if(!currentStatus) result.mandationStatus == expectedMandation else true
+        }
+        case status :: Nil => result.mandationStatus == status
+        case _ => false
+      })
     )
   }
 
-  private def retrieveIsNonMTDfBOrNonDigital(mandationStatus: HttpGetResult[MandationStatus]): Option[Boolean] = {
-    mandationStatus.fold(
-      _ => None,
-      result => Some(result.mandationStatus == nonMTDfB || result.mandationStatus == nonDigital)
-    )
-  }
+  private def retrieveIsExempt(mandationStatus: HttpGetResult[MandationStatus]): Option[Boolean] =
```

Comment:
Could we simplify these three functions into one? As far as I'm aware, at the moment in this service we treat these three statuses exactly the same (correct me if I'm wrong). Just having one function here and one boolean in the view model would simplify all this